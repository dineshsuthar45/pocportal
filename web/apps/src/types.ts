import { InputProps, TypographyProps } from "@mui/material";
import { CSSProperties, ChangeEvent } from "react";

export interface IThemeSchema {
  palette: any;
  typography: any;
  components?: any;
}

export interface IGlobalTableProps {
  row: any;
  list: any;
}

export interface IGlobalButtonProps {
  label: any;
  sublabel?: string;
  disabled?: boolean;
  onClick?: () => void;
  style?: CSSProperties;
  type?: string;
}

export interface IGlobalImgProps {
  img: string;
  alt: string;
  style?: CSSProperties;
}

export interface IGlobaltextFieldProps {
  id?: string;
  value: string;
  label: string;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  name: string;
  errmsg?: string;
  type?: string;
  disabled?: boolean;
  style?: CSSProperties;
  placeholder?: string;
  showPassword?: boolean;
  InputProps?: InputProps;
}

export interface IGlobalTypoProps {
  id?: number;
  variant: TypographyProps["variant"];
  label: string;
  sublabel?: string;
  style?: CSSProperties;
}
