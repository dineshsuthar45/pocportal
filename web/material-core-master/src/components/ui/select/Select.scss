.select-root {
	&.MuiFormControl-root {
		min-width: 80px;
		::before {
			border-bottom-color: var(--m-colour-greyScale-200);
		}

		&.dark {
			.MuiInputBase-root.MuiFilledInput-root {
				background-color: var(--m-colour-greyScale-700);
			}
			.MuiInputBase-root.MuiInput-underline {
				&::before {
					border-bottom-color: var(--m-colour-greyScale-600);
				}
				&:hover:not(.disabled):not(.menu-open) {
					&::before {
						border-bottom: 2px solid white;
						transition: border-bottom-color 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
					}
				}
			}
		}
	}

	&.error {
		.MuiSelect-select.MuiInputBase-input {
			color: var(--m-colour-red);
		}
	}

	&.disabled {
		opacity: 0.5;
		pointer-events: none;
	}

	.MuiSelect-outlined.MuiInputBase-input.MuiOutlinedInput-input {
		padding-top: 6px;
		padding-bottom: 4px;

		&.MuiInputBase-inputSizeSmall {
			padding-top: 4px;
			padding-bottom: 0px;
		}
	}

	.MuiSelect-filled.MuiInputBase-input.MuiFilledInput-input.MuiInputBase-inputSizeSmall {
		padding-top: 16px;
	}

	.MuiInputBase-root.MuiFilledInput-root {
		background-color: var(--m-colour-greyScale-50);
	}

	label {
		&.Mui-error {
			color: inherit;
		}

		&.MuiInputLabel-outlined {
			transform: translate(14px, 5px) scale(1);

			&.MuiInputLabel-sizeSmall {
				transform: translate(14px, 2px) scale(1);
			}
		}

		&[data-shrink='true'] {
			&.MuiInputLabel-outlined {
				transform: translate(14px, -6px) scale(0.85);
			}
			&.MuiInputLabel-standard {
				transform: translate(0, 0px) scale(0.85);
			}
			&.MuiInputLabel-filled {
				transform: translate(12px, 3px) scale(0.85);
			}
		}
	}

	.MuiSelect-icon {
		width: 12;
		height: 12;
	}
}
