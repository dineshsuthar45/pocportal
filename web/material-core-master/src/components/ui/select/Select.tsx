import {
  FormControl,
  FormHelperText,
  InputLabel,
  MenuItem,
  Select as MuiSelect,
  useTheme,
} from "@mui/material";
import classNames from "classnames";
import { v4 as uuidv4 } from "uuid";
import { MIconAngleDown } from "../icons";
import "./Select.scss";
import { SelectProps } from "./Select.types";

/**
 * Further properties can be found in MUI website at https://mui.com/material-ui/api/select/
 */
export const MSelect = <T extends { label: string; value: string }>({
  autoWidth = false,
  defaultOpen = false,
  disabled = false,
  displayEmpty = false,
  error = false,
  IconComponent = MIconAngleDown,
  id = uuidv4(),
  size = "small",
  variant = "standard",
  ...props
}: SelectProps<T>) => {
  const theme = useTheme();
  const { feedback, label, options, onChange } = props;

  const selectOptions = options.map((option, index) => (
    <MenuItem value={option.value} key={index}>
      {option.label}
    </MenuItem>
  ));

  const selectedValue = props.value !== undefined ? props.value.value : "";
  const labelId = `${id}-label`;

  return (
    <FormControl
      className={classNames("select-root", theme.palette.mode, {
        disabled,
        error: error,
      })}
      error={error}
      size={size}
      variant={variant}
    >
      <InputLabel id={labelId}>{label}</InputLabel>
      <MuiSelect
        {...props}
        label={label}
        labelId={labelId}
        id={id}
        value={selectedValue as "" | T | undefined}
        IconComponent={IconComponent}
        onChange={(event) =>
          onChange(
            options.find((x) => x.value === event.target.value) ?? ({} as T)
          )
        }
      >
        {selectOptions}
      </MuiSelect>
      <FormHelperText>{feedback}</FormHelperText>
    </FormControl>
  );
};

MSelect.defaultProps = {
  autoWidth: false,
  defaultOpen: false,
  disabled: false,
  displayEmpty: false,
  error: false,
  IconComponent: MIconAngleDown,
  id: uuidv4(),
  size: "small",
  variant: "standard",
};
