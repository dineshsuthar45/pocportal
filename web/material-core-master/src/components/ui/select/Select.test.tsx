import { render } from "@testing-library/react";
import React from "react";
import { MSelect } from "./Select";
import { SelectProps } from "./Select.types";

describe("Select", () => {
  interface SelectOption {
    label: string;
    value: string;
  }
  const renderComponent = () => render(<MSelect {...props} />);
  let props: SelectProps<SelectOption>;

  beforeEach(() => {
    props = {
      id: "test-id",
      options: [],
      label: "Do you agree with T&Cs?",
      onChange: () => {},
    };
  });

  it("should render correctly", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it("render with options", () => {
    props.options = [
      {
        label: "Yes",
        value: "1",
      },
      {
        label: "No",
        value: "0",
      },
    ];
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
