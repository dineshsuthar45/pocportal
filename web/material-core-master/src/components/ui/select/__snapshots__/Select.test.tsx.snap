// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select render with options 1`] = `
<div>
  <div
    class="MuiFormControl-root select-root light css-1nrlq1o-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-standard css-z5e0z9-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      id="test-id-label"
    >
      Do you agree with T&Cs?
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall  css-m5hdmq-MuiInputBase-root-MuiInput-root-MuiSelect-root"
      options="[object Object],[object Object]"
    >
      <div
        aria-controls=":r1:"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="test-id-label test-id"
        class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input MuiInputBase-inputSizeSmall css-jd1zyo-MuiSelect-select-MuiInputBase-input-MuiInput-input"
        id="test-id"
        role="combobox"
        tabindex="0"
      >
        <span
          class="notranslate"
        >
          ​
        </span>
      </div>
      <input
        aria-hidden="true"
        aria-invalid="false"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value=""
      />
      <svg
        class="MuiSelect-icon MuiSelect-iconStandard css-19xtc65-MuiSelect-icon"
        fill="#6F7880"
        height="10"
        viewBox="0 0 10 10"
        width="10"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m.5 3 4.1 4.1c.2.2.3.3.4.3s.2 0 .4-.2L9.6 3a.608.608 0 0 0 0-.8.608.608 0 0 0-.8 0L5 6.1 1.2 2.3a.483.483 0 0 0-.7 0 .483.483 0 0 0 0 .7Z"
        />
      </svg>
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-sizeSmall css-u6loev-MuiFormHelperText-root"
    />
  </div>
</div>
`;

exports[`Select should render correctly 1`] = `
<div>
  <div
    class="MuiFormControl-root select-root light css-1nrlq1o-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-standard css-z5e0z9-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      id="test-id-label"
    >
      Do you agree with T&Cs?
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall  css-m5hdmq-MuiInputBase-root-MuiInput-root-MuiSelect-root"
      options=""
    >
      <div
        aria-controls=":r0:"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="test-id-label test-id"
        class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input MuiInputBase-inputSizeSmall css-jd1zyo-MuiSelect-select-MuiInputBase-input-MuiInput-input"
        id="test-id"
        role="combobox"
        tabindex="0"
      >
        <span
          class="notranslate"
        >
          ​
        </span>
      </div>
      <input
        aria-hidden="true"
        aria-invalid="false"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value=""
      />
      <svg
        class="MuiSelect-icon MuiSelect-iconStandard css-19xtc65-MuiSelect-icon"
        fill="#6F7880"
        height="10"
        viewBox="0 0 10 10"
        width="10"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m.5 3 4.1 4.1c.2.2.3.3.4.3s.2 0 .4-.2L9.6 3a.608.608 0 0 0 0-.8.608.608 0 0 0-.8 0L5 6.1 1.2 2.3a.483.483 0 0 0-.7 0 .483.483 0 0 0 0 .7Z"
        />
      </svg>
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-sizeSmall css-u6loev-MuiFormHelperText-root"
    />
  </div>
</div>
`;
